itest.py

```diff
@@ -0,0 +1,832 @@
+    logs = build_container.logs(stream=True)
+    for line in logs:
+        print(line.decode(), end='')
+    res = build_container.wait()
+    logger.debug(f'Container result of build_image={build_image!r}, build_container={res!r}')
+
+    return res['StatusCode']
+
+def ask_build_plugin():
+    # docker run --rm -v /home/marflo/Documents/iRODS/irods_microservice_plugins_curl:/irods_plugin_source:ro -v /home/marflo/Documents/iRODS/build-artifacts/ubuntu-22:/irods_packages:ro -v /home/marflo/Documents/iRODS/build-artifacts/ubuntu-22/curl_build:/irods_plugin_build -v /home/marflo/Documents/iRODS/build-artifacts/ubuntu-22/curl:/irods_plugin_packages plugin-builder:ubuntu-22 --build_directory /irods_plugin_build
+    client = docker.from_env()
+
+    # Get valid images
+    yes_images = tuple(get_valid_tags(client, IRODS_PLUGIN_BUILDER_NAME))
+    assert(len(yes_images) > 0)
+
+    for image in yes_images:
+        print(image)
+
+    # Prompt user for image choice
+    os_choice = input("What image ya wanna run? ")
+    validate_choice(os_choice, yes_images)
+
+    # Get the selected image
+    build_image = client.images.get(f'{IRODS_PLUGIN_BUILDER_NAME}:{os_choice}')
+
+    # Gather valid plugins
+    yes_plugins = tuple(get_supported_plugins())
+    assert(len(yes_plugins) > 0)
+
+    for plugin in yes_plugins:
+        print(plugin)
+    
+    # Prompt user for desired plugin
+    plugin_choice = input("What plugin ya wanna build? ")
+    validate_choice(plugin_choice, yes_plugins)
+
+    # Dumb trim of plugin
+    for prefix in PLUGIN_PREFIXES:
+        if plugin_choice.startswith(prefix):
+            plugin_name = plugin_choice[len(prefix)+1:]
+
+    # Gather args, convert PurePaths to Paths
+    dirs_to_format = [IRODS_PLUGIN_SRC_DIR, IRODS_PACKAGES_DIR, IRODS_PLUGIN_BUILD_DIR, IRODS_PLUGIN_PACKAGE_DIR]
+    dirs_to_mount = [d for d in format_pure_path(dirs_to_format, os_name=os_choice, full_plugin_name=plugin_choice, short_plugin_name=plugin_name)]
+
+    # Create directories if needed
+    for d in (d for d in dirs_to_mount if not d.exists()):
+        logger.info(f'The path=[{d!r}] does not exist, creating.')
+        d.mkdir(parents=True)
+
+    # Create symlinks for build & testing
+    sym_link_needed = [dirs_to_mount[1], dirs_to_mount[3]]
+    for d in sym_link_needed:
+        supported_oses = {'ubuntu-22.04': 'Ubuntu_22',
+                          'ubuntu-24.04': 'Ubuntu_24',
+                          'debian-11': 'Debian gnu_linux_11',}
+        if os_choice in supported_oses:
+            link_dir = d / supported_oses[os_choice]
+            
+            # Assume if exists is link
+            if not link_dir.exists():
+                logger.info(f'The link_dir=[{link_dir!r}] does not exist, creating.')
+                link_dir.symlink_to('.', target_is_directory=True)
+        else:
+            logger.warning(f'The os_choice=[{os_choice!r}] is not supported!')
```

Comment:
symlink code will need to be removed if/when it being a requirement for the plugins are gone.

Reply:
This applies throughout the code (in case it pops up elsewhere).

Reply:
Updated symlink code, removed when to needed.